src/DynamoCore/Logging/DynamoAnalyticsClient.cs

```diff
@@ -165,7 +178,7 @@ public DynamoAnalyticsClient(DynamoModel dynamoModel)
         /// </summary>
         public void Start()
         {
-            if (ReportingAnalytics)
+            if (preferences!= null && preferences.IsAnalyticsReportingApproved)
```

Comment:
@QilongTang on the line below should the reference to `ReportingAnalytics` also be changed to `preferences.IsAnalyticsReportingApproved` because the service still might not be initialized yet. What do you think?

Reply:
@mjkkirschner Yup, makes a lot of sense! I will draft a PR soon